pkg/openlambda/ol.go

```diff
@@ -0,0 +1,163 @@
+package openlambda
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/serverlessresearch/srk/pkg/srk"
+)
+
+type olConfig struct {
+	// Command to run base openlambda manager ('ol')
+	cmd string
+	// Working directory for openlambda
+	dir string
+	// Keeps track of if we've started an ol worker or not
+	sessionStarted bool
+}
+
+func NewConfig(cmd string, dir string) srk.FaasService {
+	return &olConfig{cmd, dir, false}
+}
+
+func (self *olConfig) Package(rawDir string) (string, error) {
+	tarPath := filepath.Clean(rawDir) + ".tar.gz"
+	rerr := tarRaw(rawDir, tarPath)
+	if rerr != nil {
+		return "", rerr
+	}
+	return tarPath, nil
+}
+
+func (self *olConfig) Install(rawDir string) error {
+	tarPath := filepath.Clean(rawDir) + ".tar.gz"
+
+	installPath := filepath.Join(self.dir, "registry", filepath.Base(tarPath))
+	//OL is managed by root so we have to use sudo commands for everything
+	cmd := exec.Command("sudo", "sh", "-c", "cp "+tarPath+" "+installPath)
+	if out, err := cmd.CombinedOutput(); err != nil {
+		fmt.Printf("Failed to install function: %v\n", err)
+		fmt.Printf(string(out))
+		return err
+	}
+	fmt.Println("Open Lambda function installed to: " + installPath)
+	return nil
+}
+
+func (self *olConfig) Destroy() {
+	if self.sessionStarted {
+		self.terminateOlWorker()
+	}
+}
+
+func (self *olConfig) Invoke(fName string, args string) (resp *bytes.Buffer, rerr error) {
+	if !self.sessionStarted {
+		self.launchOlWorker()
```

Comment:
We might want a separate command to launch the worker / boot up the OpenLambda environment. Consider how it works with AWS - Lambda is running in the background all the time and maintains cached function instances. The invoke command just runs a function.